macaw-aarch32/src/Data/Macaw/ARM/Arch.hs

```diff
@@ -652,22 +652,20 @@ a32InstructionMatcher (ARMDis.Instruction opc operands) =
       ARMDis.SVC_A1 ->
         case operands of
           ARMDis.Bv4 _opPred ARMDis.:< ARMDis.Bv24 imm ARMDis.:< ARMDis.Nil -> Just $ do
-            let r0 = ARMReg.ARMGlobalBV (ASL.knownGlobalRef @"_R0")
-            let r1 = ARMReg.ARMGlobalBV (ASL.knownGlobalRef @"_R1")
-            sc <- ARMSyscall imm <$> G.getRegVal r0
-                                 <*> G.getRegVal r1
-                                 <*> G.getRegVal (ARMReg.ARMGlobalBV (ASL.knownGlobalRef @"_R2"))
-                                 <*> G.getRegVal (ARMReg.ARMGlobalBV (ASL.knownGlobalRef @"_R3"))
-                                 <*> G.getRegVal (ARMReg.ARMGlobalBV (ASL.knownGlobalRef @"_R4"))
-                                 <*> G.getRegVal (ARMReg.ARMGlobalBV (ASL.knownGlobalRef @"_R5"))
-                                 <*> G.getRegVal (ARMReg.ARMGlobalBV (ASL.knownGlobalRef @"_R6"))
-                                 <*> G.getRegVal (ARMReg.ARMGlobalBV (ASL.knownGlobalRef @"_R7"))
+            sc <- ARMSyscall imm <$> G.getRegVal ARMReg.r0
+                                 <*> G.getRegVal ARMReg.r1
+                                 <*> G.getRegVal ARMReg.r2
+                                 <*> G.getRegVal ARMReg.r3
+                                 <*> G.getRegVal ARMReg.r4
+                                 <*> G.getRegVal ARMReg.r5
+                                 <*> G.getRegVal ARMReg.r6
+                                 <*> G.getRegVal ARMReg.r7
             res <- G.addExpr =<< evalArchFn sc
             -- res is a tuple of form (R1, R0).  This is reversed from the
             -- user provided return Assignment of empty :> R0 :> R1 because
             -- the conversion from Assignment to Tuple reversed the order.
-            G.setRegVal r1 =<< G.addExpr (G.AppExpr (MC.TupleField PC.knownRepr res PL.index0))
-            G.setRegVal r0 =<< G.addExpr (G.AppExpr (MC.TupleField PC.knownRepr res PL.index1))
+            G.setRegVal ARMReg.r1 =<< G.addExpr (G.AppExpr (MC.TupleField PC.knownRepr res PL.index0))
+            G.setRegVal ARMReg.r0 =<< G.addExpr (G.AppExpr (MC.TupleField PC.knownRepr res PL.index1))
 
             -- Increment the PC; we don't get the normal PC increment from the
             -- ASL semantics, since we are intercepting them to just add this statement
```

Comment:
GitHub won't let me highlight it for some reason, but there is an occurrence of `ARMReg.ARMGlobalBV (ASL.knownGlobalRef @"_PC")` just below this line that could be shortened to `ARMReg.pc`.